src/App.tsx

```diff
@@ -1,14 +1,55 @@
 /* eslint-disable max-len */
-import React from 'react';
+import { useEffect, useState, FC } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
-export const App: React.FC = () => {
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [preparedTodos, setPreparedTodos] = useState<Todo[]>([]);
+  const [sorted, setSorted] = useState('');
+  const [query, setQuery] = useState('');
+  const [modalTodo, setModalTodo] = useState<Todo | undefined>();
+
+  useEffect(() => {
+    getTodos().then(data => {
+      setTodos(data);
+      setPreparedTodos(data);
+    });
+  }, []);
+
+  useEffect(() => {
+    const preparingTodos = () => {
+      if (sorted === 'active') {
+        setPreparedTodos(todos.filter(todo => !todo.completed && todo.title.includes(query.toLowerCase())));
+      } else if (sorted === 'completed') {
+        setPreparedTodos(todos.filter(todo => todo.completed && todo.title.includes(query.toLowerCase())));
+      } else {
+        setPreparedTodos(todos.filter(todo => todo.title.includes(query.toLowerCase())));
```

Comment:
you forgot to add `todo.title.toLowerCase()` to fix the test
<img width="528" alt="image" src="https://github.com/mate-academy/react_dynamic-list-of-todos/assets/70220667/b4de69ca-89c8-4b5c-8f1d-cf769db55c1f">